servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/DefaultGrpcClientMetadata.java

```diff
@@ -31,37 +32,94 @@
         this(path, (GrpcExecutionStrategy) null);
     }
 
+    /**
+     * Creates a new instance which uses the provided path and content codec, the default execution strategy,
+     * and no deadline.
+     *
+     * @param path for the associated <a href="https://www.grpc.io">gRPC</a> method.
+     * @param requestEncoding {@link ContentCodec} to use for the associated <a href="https://www.grpc.io">gRPC</a>
+     * method.
+     */
     protected DefaultGrpcClientMetadata(final String path, final ContentCodec requestEncoding) {
         this(path, null, requestEncoding);
     }
 
     /**
-     * Creates a new instance.
+     * Creates a new instance which uses the provided path, content codec and deadline and the default execution
+     * strategy.
+     *
+     * @param path for the associated <a href="https://www.grpc.io">gRPC</a> method.
+     * @param requestEncoding {@link ContentCodec} to use for the associated <a href="https://www.grpc.io">gRPC</a>
+     * method.
+     * @param deadline A timeout deadline after which the response is no longer wanted.
+     */
+    protected DefaultGrpcClientMetadata(final String path,
+                                        final ContentCodec requestEncoding,
+                                        final Duration deadline) {
+        this(path, null, requestEncoding, deadline);
+    }
+
+    /**
+     * Creates a new instance which uses the provided path and execution strategy and the default content codec
+     * (identity) and no deadline.
      *
      * @param path for the associated <a href="https://www.grpc.io">gRPC</a> method.
      * @param strategy {@link GrpcExecutionStrategy} to use for the associated <a href="https://www.grpc.io">gRPC</a>
      * method.
      */
     protected DefaultGrpcClientMetadata(final String path,
                                         @Nullable final GrpcExecutionStrategy strategy) {
-        this(path, strategy, identity());
+        this(path, strategy, identity(), null);
     }
 
+    /**
+     * Creates a new instance which uses the provided path, content codec and the default execution
+     * strategy and no deadline.
+     *
+     * @param path for the associated <a href="https://www.grpc.io">gRPC</a> method.
+     * @param strategy {@link GrpcExecutionStrategy} to use for the associated <a href="https://www.grpc.io">gRPC</a>
+     * method.
+     * @param requestEncoding {@link ContentCodec} to use for the associated <a href="https://www.grpc.io">gRPC</a>
+     * method.
+     */
     protected DefaultGrpcClientMetadata(final String path,
                                         @Nullable final GrpcExecutionStrategy strategy,
                                         final ContentCodec requestEncoding) {
+        this(path, strategy, requestEncoding, null);
+    }
+
+    /**
+     * Creates a new instance which uses the provided path, content codec and deadline and the default execution
+     * strategy.
```

Comment:
Hard to keep consistency and correctness between `@param` and initial line when there are many overloads and code changes over time. For example, this method doesn't use "default execution strategy", it takes the execution strategy from the user. So, the javadoc is already not consistent with the params. Consider reverting back to simple "Creates a new instance". Params give reasonable explanation about the differences between overloads.

Reply:
fair point.